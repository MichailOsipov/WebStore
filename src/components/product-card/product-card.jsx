import * as React from 'react';
import {
    Card,
    CardActionArea,
    CardMedia,
    CardContent,
    Typography,
    CardActions
} from '@material-ui/core';

export const ProductCard = ({name, imageSrc, actions}) => (
    <Card style={{maxWidth: 345}}>
        <CardActionArea>
            <CardMedia
                style={{height: 140}}
                title={`product-${name}`}
                image={imageSrc}
            />
            <CardContent>
                <Typography variant="h5" component="h2">
                    {name}
                </Typography>
            </CardContent>
        </CardActionArea>
        {actions && (
            <CardActions>
                {actions}
            </CardActions>
        )}
    </Card>
);
