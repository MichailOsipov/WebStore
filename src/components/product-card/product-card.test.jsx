import * as React from 'react';
import {shallow} from 'enzyme';
import {ProductCard} from './product-card';

describe('ProductCard', () => {
    test('rendering', () => {
        const wrapper = shallow(
            <ProductCard
                name="name"
                imageSrc="imageSrc"
                actions={(
                    <div>Actions</div>
                )}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
