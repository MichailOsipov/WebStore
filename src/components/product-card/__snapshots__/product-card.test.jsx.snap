// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard rendering 1`] = `
<WithStyles(Card)>
  <WithStyles(CardActionArea)>
    <WithStyles(CardMedia)
      image="imageSrc"
      style={
        Object {
          "height": 140,
        }
      }
      title="product-name"
    />
    <WithStyles(CardContent)>
      <WithStyles(Typography)
        component="h2"
        variant="h5"
      >
        name
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
  </WithStyles(CardActionArea)>
  <WithStyles(CardActions)>
    <div>
      Actions
    </div>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;
